tests/compile-errors/src/different-serializable-type.stderr

```diff
@@ -0,0 +1,10 @@
+error[E0308]: mismatched types
+  --> $DIR/different-serializable-type.rs:18:1
+   |
+18 | #[usdt::provider]
+   | ^^^^^^^^^^^^^^^^^ expected struct `Expected`, found struct `Different`
+...
+25 |     my_provider_my_probe!(|| Different { x: 0 });
```

Comment:
I'm not sure how tenable it is, but it would be cool if this underlined `Different { x: 0 }`

Reply:
Also deferring this one. It'd be nice, but I don't think we have any way to get the span of the closure from our proc macros.